src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -1728,11 +1728,6 @@ internal sealed override void AddSynthesizedAttributes(PEModuleBuilder moduleBui
                         PEModule.ByRefLikeMarker, nameof(CompilerFeatureRequiredFeatures.RefStructs), isOptionalUse: true);
                 }
             }
-            else if (this.IsExtension)
-            {
-                addPoisonAttributes(ref attributes, compilation, hasObsolete,
```

Comment:
@AlekseyTs @jjonescz I think it's worth calling out that this change also removes `[Obsolete(ExtensionMarker)]`. This does not affect downlevel support, but is meant to prevent/protect older compilers. 
But since we'll codegen normal types with static methods, I think the attribute is no longer beneficial.

Reply:
Then I guess this part of the speclet should be removed:

> The type is marked with Obsolete and CompilerFeatureRequired attributes.

Reply:
Will do